pandas/_libs/tslibs/offsets.pyx

```diff
@@ -254,7 +254,7 @@ cdef _to_dt64D(dt):
     if getattr(dt, 'tzinfo', None) is not None:
         # Get the nanosecond timestamp,
         #  equiv `Timestamp(dt).value` or `dt.timestamp() * 10**9`
-        naive = dt.astimezone(None)
+        naive = dt.replace(tzinfo=None)
```

Comment:
One more item: Could you write a comment above this line linking the original Github issue and describing that `naive` should be the naive wall time instead of the naive absolute time of `dt`

Reply:
Sure, I will do it

Like that?

```python
    if getattr(dt, 'tzinfo', None) is not None:
        # Get the nanosecond timestamp,
        #  equiv `Timestamp(dt).value` or `dt.timestamp() * 10**9`
        # The `naive` must be the `dt` naive wall time
        #  instead of the naive absolute time (GH#49441)
        naive = dt.replace(tzinfo=None)
        dt = np.datetime64(naive, "D")
    else:
```